File: CodeCheck/All Checks/Language Specific/Java/JAVA_42/test01.java
public class Test01 {
    public static void main(String[] args) {
        int i, j, k;
        int sum = 0;

        if (sum == (0)) {
            sum += 1;
            if (sum == (5))
                sum += 3;
            else if (sum == (2))
                sum += 5;
            else
                sum += 6;
         }                        // UNDCC_Violation
        else if (sum == (1)) {
            sum += 4;
       }                          // UNDCC_Violation
        else {
            sum += 7;
        }                         // UNDCC_Valid

        for (i = 0; i < (5); i++) {
            sum += i;
            for (j = 0; j < (5); j++) {
                sum += j;
                for (k = 0; k < (5); k++)
                    sum += k;
             }                    // UNDCC_Violation
         }                        // UNDCC_Violation

        for (i = 0; i < (5); i++)
            sum += i;

        i = 0;
        while (i < (5)) {
            i += 1;
            sum += i;
         }                        // UNDCC_Violation

        i = 0;
        while (i < (5))
            i += 1;

        i = 0;
        do {
            sum += i;
            i++;
            if (sum == (1))
                sum += 4;
         } while (i < (5));       // UNDCC_Violation

        i = 0;
        do
            i++;
        while (i < (5));
    }                             // UNDCC_Valid
}                                 // UNDCC_Valid

File: CodeCheck/Published Standards/AUTOSAR/A5-2-5A/test01.cpp
// Custom

#include <iostream>

int main()
{

  int arr1[10];   /* UndCC_Valid */
  arr1[0] = 2;

  int n = 10;
  int arr2[n];    /* UndCC_Violation - identifier used to size array */
  arr2[0] = 2;

  int arr3[6] = { 10, 20, 30, 40 };    /* UndCC_Valid */

  printf("%d %d %d", arr1[0], arr2[0], arr3[0]);

  return 0;
}
